packages/node/src/methods/state-channel/create/controller.ts

```diff
@@ -18,62 +19,80 @@ const CREATE_PROXY_AND_SETUP_GAS = 6e6;
  * having been opened via the creation of the multisig.
  * In "creating a channel", this also creates a multisig while sending details
  * about this multisig to the peer with whom the multisig is owned.
+ *
+ * Details: This returns the hash of the multisig deployment transaction so the
+ * caller can async listen to the on chain confirmation of the transaction being
+ * mined. Once the actual multisig is deployed, its _address_ is sent as an event
+ * to the caller via the `NODE_EVENTS.CREATE_CHANNEL` event. This is because
+ * the address of the multisig is not retrievable from the transaction hash
+ * since the multisig is deployed through an internal transaction of a proxy
+ * factory contract.
  * @param params
  */
 export default async function createChannelController(
   requestHandler: RequestHandler,
   params: Node.CreateChannelParams
-): Promise<Node.CreateChannelResult> {
-  const multisigAddress = await deployMinimumViableMultisigAndGetAddress(
+): Promise<Node.CreateChannelTransactionResult> {
+  const {
+    proxyFactory,
+    transactionHash
+  } = await deployMinimumViableMultisigAndGetTransactionHash(
     params.owners,
     requestHandler.wallet,
     requestHandler.networkContext.MinimumViableMultisig,
     requestHandler.networkContext.ProxyFactory
   );
 
-  const [respondingXpub] = params.owners.filter(
-    owner => owner !== requestHandler.publicIdentifier
-  );
-
-  const stateChannelsMap = await requestHandler.instructionExecutor.runSetupProtocol(
-    {
+  proxyFactory.once("ProxyCreation", async multisigAddress => {
+    const payload = {
       multisigAddress,
-      respondingXpub,
-      initiatingXpub: requestHandler.publicIdentifier
-    }
-  );
+      owners: params.owners
+    } as Node.CreateChannelResult;
 
-  await requestHandler.store.saveStateChannel(
-    stateChannelsMap.get(multisigAddress)!
-  );
+    const [respondingXpub] = params.owners.filter(
+      owner => owner !== requestHandler.publicIdentifier
+    );
```

Comment:
Might be nicer if the params are `initiatingXpub` and `respondingXpub` or, more generally speaking, clearly specifying who "I am" and who "I want to open a channel with is" versus passing in an array of the owners. Here's Raiden's API: https://raiden-network.readthedocs.io/en/v0.9.0/api_walkthrough.html#opening-a-channel. They use `partner_address` and get "my address" from the rest of the message.

Reply:
This would be nice (though it would limit the public facing API to two parties which would be awkward in terms of API versioning - can discuss more later).